BBallStatsTrack/src/com/example/bballstatstrack/models/Game.java

```diff
@@ -0,0 +1,496 @@
+package com.example.bballstatstrack.models;
+
+import java.util.Date;
+import java.util.UUID;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.example.bballstatstrack.models.game.GameLog;
+import com.example.bballstatstrack.models.gameevents.BlockEvent;
+import com.example.bballstatstrack.models.gameevents.GameEvent;
+import com.example.bballstatstrack.models.gameevents.GameEvent.ReboundType;
+import com.example.bballstatstrack.models.gameevents.GameEvent.ShotType;
+import com.example.bballstatstrack.models.gameevents.ReboundEvent;
+import com.example.bballstatstrack.models.gameevents.ShootEvent;
+import com.example.bballstatstrack.models.gameevents.TurnoverEvent;
+import com.example.bballstatstrack.models.gameevents.foulevents.NonShootingFoulEvent;
+import com.example.bballstatstrack.models.gameevents.foulevents.ShootingFoulEvent;
+import com.example.bballstatstrack.models.utils.GameEventDeserializer;
+
+import android.text.format.DateFormat;
+import android.util.Log;
+import android.util.SparseArray;
+
+public class Game
+{
+    private static final String B_BALL_STAT_TRACK = "BBallStatTrack";
+
+    private static final int MAX_SHOT_CLOCK = 24;
+
+    private Team mAwayTeam;
+
+    private Team mHomeTeam;
+
+    private Team mHasBallPossession;
```

Comment:
mHasBallPossession is used specifically as a reference to both mAwayTeam or mHomeTeam. I implemented it in Game instead of inside of Team because I think the Team object doesn't really care if it has ball possession or not. The Game decides who has the ball. I also implemented it this way so that it would be impossible to have both teams being in ball possession. However, if you have more ideas, please feel free to share them.